### DISCUSSION TITLE
M5 Stick Is Unavailable

### DISCUSSION THREAD

**Discussion by jorgediazdev** [PlummersSoftwareLLC/NightDriverStrip#628] [Q&A]
Dave mentions to use the M5 stick, but it is currently unavailable. Would a regular ESP32 Module work or does it have to be an M5 Stick? I noticed that the M5 is mentioned specifically within the code base.

**Comment by robertlipe** [comment]
We support lots of devices. The pin outs may change a little. Not too long
ago, I made a big post on recommendations. Look for that in the archives



 The Adafruit Feather and the Lilygo devices are also on the list for big
names. Definitely pay the extra coins for devices with at least 2mb of
PSRAM.

You may have to remix the combination of effects and board in config.h as
they're not really well separated, but that's easy enough. Just start with
demo to get the board basics worked out then replace the demo with the
appropriate effects modules.

I inventory bare yd-esp32-s3 N16R8 boards for this and similar projects.
Socketing them has had me looking at the mini s3s just because the sockets
cost almost at much at the board does at 44 pins.

**Comment by rbergen** [comment]
In addition to @robertlipe's excellent answer: there is a successor to the M5StickC Plus that Dave mentioned that is available, which is somewhat predictably named the M5StickC Plus2. 

At this time, the M5StickC Plus2 doesn't work with this project yet (as documented in issue #618), partly because it first needs to be supported by one of our dependencies - all of this is open source, so that is how it works. Once that's been sorted out and we can get our hands on some, we'll try to get the Plus2 working as well. I for one like the "non-2" M5StickC Plus form factor too.

**Comment by robertlipe** [comment]
I saw part of a video <https://www.youtube.com/watch?v=gHsJP5br_ok> last
night (that seemed like a rerun, but it's contemporary...) that mentioned
the HiLetGo board in a similar context.

It has a screen (more GPIO pins occupied, cost that might not make sense in
an attic installation, .etc.) and a weird reference to 8266 that I couldn't
quite work out. It looks like HiLetgo is a reseller and the pictured board
is actually a  Heltec board. Copying the referral link from that video:
https://amzn.to/3xfaLrP
<https://www.youtube.com/redirect?event=video_description&redir_token=QUFFLUhqbTJScHllS1Q3ZzhQMXFlZXlOLTBfbk1GNzA5Z3xBQ3Jtc0trSGFSU1FnYUVKWmdmdVFmTi1WUDA0R29ESXVwZkp5RVJaLTAxZUVBUjZwMFpic1d6TWJ1Q2F1Z01qU2RIQ3AyeU5KZmFhc0E0WTdZZHluNDNZTGl2SEpzMkQzSEJyMmsxX29lNlAzTmc4SmdGRnhBQQ&q=https%3A%2F%2Famzn.to%2F3xfaLrP&v=gHsJP5br_ok>
It
seems likely that at least one member of the active development group has
one of those boards. :-)

Honestly, it's a highly cost-competitive market and a lot of resellers
don't really know what they're reselling and there's often a language
barrier, so it can be really frustrating to figure out what's what.

As Rutger alludes, I don't think any one person has one of everything and
projects like this just kind of rely upon people caring enough about
particular combinations to buy them, make them work, and share and report
upon their findings. When an Asian market decides it can save a nickel on a
product and change the device in an incompatible way, it can put the whole
open source biz in a spin for a bit. That seems to be what's going on with
the M5's right now. There's the discontinued ones and the ones that don't
work yet. - which is really odd given how simple these devices really are.
(They're almost all a module plunked onto a reference design and then fit
with slightly different on-board accessories and packaging.)
